central/imageintegration/datastore/singleton.go

```diff
@@ -22,44 +23,33 @@ var (
 	ad DataStore
 )
 
-func initializeDefaultIntegrations(ctx context.Context, storage store.Store) {
-	integrations, err := ad.GetImageIntegrations(ctx, &v1.GetImageIntegrationsRequest{})
+func initializeDefaultIntegrations(storage store.Store) {
+	ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowAllAccessScopeChecker())
+	iis, err := storage.GetAll(ctx)
 	utils.CrashOnError(err)
-	if !env.OfflineModeEnv.BooleanSetting() && len(integrations) == 0 {
+	if !env.OfflineModeEnv.BooleanSetting() && len(iis) == 0 {
 		// Add default integrations
 		for _, ii := range store.DefaultImageIntegrations {
 			utils.Must(storage.Upsert(ctx, ii))
 		}
 	}
 }
 
-func initializeManager(ctx context.Context) {
-	// Initialize the integration set with all present integrations.
-	integrationManager := enrichment.ManagerSingleton()
-	integrations, err := ad.GetImageIntegrations(ctx, &v1.GetImageIntegrationsRequest{})
-	if err != nil {
-		log.Errorf("unable to use previous integrations: %s", err)
-	}
-	for _, ii := range integrations {
-		if err := integrationManager.Upsert(ii); err != nil {
-			log.Errorf("unable to use previous integration %s: %v", ii.GetName(), err)
-		}
-	}
-}
-
 func initialize() {
 	// Create underlying store and datastore.
 	var storage store.Store
+	var indexer index.Indexer
+
 	if features.PostgresDatastore.Enabled() {
 		storage = postgres.New(globaldb.GetPostgres())
+		indexer = postgres.NewIndexer(globaldb.GetPostgres())
 	} else {
 		storage = bolt.New(globaldb.GetGlobalDB())
+		indexer = index.New(globalindex.GetGlobalTmpIndex())
 	}
-	ad = New(storage)
-
-	ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowAllAccessScopeChecker())
-	initializeDefaultIntegrations(ctx, storage)
-	initializeManager(ctx)
+	initializeDefaultIntegrations(storage)
+	searcher := search.New(storage, indexer)
+	ad = New(storage, indexer, searcher)
```

Comment:
an we change the name for this? I don't know what `ad` was meant to mean originally, so now might be a good time to give it a better name